client/src/plugins/element-templates-modal/modeler/EditorActions.js

```diff
@@ -37,6 +37,10 @@ export default class EditorActions {
       return selectedElements[ 0 ];
     });
 
+    editorActions.register('getValidElementTemplates', () => {

```

Comment:
Could you elaborate what we want to achieve and how that is the proper solution? Do we already use editor actions in such a manner?

I believe you'd like to retrieve the valid templates in a contextual manner, from within a UI component, right?

```
React Component -> Contextual Get Diagram service -> Interact with it
```

To accomplish that, from the architectural point of view I'd assume we have a mechanism that provides us with a the diagram context (if any) and interact with it. 

The modal could be instantiated with that context right away, too.

Reply:
**What I want to achieve**
Use the elementTemplate registry (which already loaded and did validity filtering of element templates) rather than manually doing the element template retrieval and validity filtering.

**How I implemented it**
We don't have direct access to the Modeler instance from the Modal ReactComponent. However, when loading the Modal (which is a plugin), we also load `EditorActions.js`. `EditorActions` already inject other services:
```
EditorActions.$inject = [
  'commandStack',
  'editorActions',
  'selection'
];
```

I added `elementTemplates` on top. So I am re-using the existing mechanism.

Reply:
I was re-using the exsting `editorActions` and concepts there. 

I will take a look whether I find a better way to not registerActions over `EditorActions`, but e.g., directly with diagram context. However, IMO this PR follows the existing architecture, so switching away from that might be worth another PR.

Reply:
I agree that the what you're proposing is a substantial change that is worth shipping, I simply have the troubles with the _how_.

If we interpret the actions in the broadest possible way we could indeed argue that retrieving anything from the current editor is an editor action. However, we have to be careful how many additional things we'd like to feed into the existing concept. Editor actions from the bpmn-js point of view are user-triggered operations on the diagram, not generic data retrival operations. Editor actions from the Modeler point of view may have been used for different things in the mean time...

If "doing something in the context of the current editor" is a concern, I'd consider two alternatives:

* A generic `exec` action that invokes a function in the context of the given editor:
    ```javascript
    triggerAction('exec', function(elementTemplates) {
      return 'Foo';
    })
    ```
* A generalized way to access and inject the diagram context into a UI component

Just proposing, feel free to discard.

Reply:
:+1: :+1: :+1: